### ISSUE TITLE
False negative in Initialization Checker with circular dependencies between static fields

### ISSUE THREAD

**Issue by flo2702** [eisop/checker-framework#1181]
The Nullness Checker / Initialization Checker does not correctly handle initialization of static fields in the case where two classes circularly depend on each other.

For example, the following program throws a `NullPointerException`, which is not detected by the Checker Framework.

```java
public class Test {
   static Object f = Other.g;
   public static void main(String[] args) {
       System.out.println(Other.f.toString());
   }
}

class Other {
  static Object f = Test.f;
  static Object g = "abc";
}
```
